### PR TITLE
cstate/pstate: Skip check on non intel arches

### PR DESCRIPTION
**Pull request by jschintag**
Intel driver is not available on other arches, skip check.

Signed-off-by: Jan Schintag <jan.schintag@de.ibm.com>

### CODE REVIEW

**Comment by marquiz** [comment]
#570 addresses the logging issues.

We can disable it altogether, but I think it would be bette to use conditional build with filename suffixes, like we do for the other features (like cpuid). And, do the same for pstate as well. I.e. have files `cstate_amd64.go` and `cstate_stub.go` (and pstate similarly).

**Comment by jschintag** [comment]
@marquiz I'll rebase and make the changes after #570 is merged.
/hold

**Comment by marquiz** [comment]
> @marquiz I'll rebase and make the changes after #570 is merged.

Perfect, thanks!

**Review by marquiz — APPROVED** [review]
Fast response, @jschintag 👍  At a quick glance looks good to me
/ok-to-test

File: source/cpu/cstate_amd64.go
Comment (marquiz) [inline source/cpu/cstate_amd64.go:2]: nit: redundant in the file suffixed `_amd64`, could be dropped

File: source/cpu/pstate_amd64.go
Comment (marquiz) [inline source/cpu/pstate_amd64.go:2]: nit: redundant in the file suffixed `_amd64`, could be dropped

**Review by marquiz — CHANGES_REQUESTED** [review]
Actually, found one small nit

**Comment by jschintag** [comment]
Yeah, i forgott to remove `runtime` from pstate_amd64.go
Regarding the build flags, i copied them from rdt. I can leave them out as well.

**Comment by marquiz** [comment]
> Yeah, i forgott to remove `runtime` from pstate_amd64.go

OK

> Regarding the build flags, i copied them from rdt. I can leave them out as well.

Yeah, please remove (as e.g. `source/cpu/power_amd64.go` does not have that). We should prolly remove those from rdt, too (in a separate PR)

**Review by marquiz — APPROVED** [review]
Thanks @jschintag for this optimization 👍 

/lgtm

**Pull request merged**
